src/shell/shell.cpp

```diff
@@ -489,14 +489,17 @@ void DOS_Shell::RunInternal()
 
 void DOS_Shell::Run()
 {
-	char input_line[CMD_MAXLINE] = {0};
-	std::string line;
-	if (cmd->FindExist("/?", false) || cmd->FindExist("-?", false)) {
+	// COMMAND.COM's /C and /INIT spawn sub-commands. When parsing help, we need
+	// to be sure the /? and -? are intended for us and not part of the
+	// sub-command.
+	if (cmd->ExistsPriorTo({"/?", "-?"}, {"/C", "/INIT"})) {
```

Comment:
Nifty fix! And now the behaviour makes complete sense! 😅

Reply:
Thanks for the quick check and review, @johnnovak !